test/integration/base.py

```diff
@@ -0,0 +1,113 @@
+import contextlib
+import errno
+import os
+import socket
+import subprocess
+import tempfile
+import unittest
+
+
+class PelikanTest(unittest.TestCase):
+    reserved_ports = set()
+
+    @staticmethod
+    def default_path():
+        return os.path.realpath(os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            '../../_build/_bin'
+        ))
+
+    # from http://stackoverflow.com/a/22106569
+    @staticmethod
+    def get_open_port(lowest_port = 10000, highest_port = None, bind_address = '', *socket_args, **socket_kwargs):
```

Comment:
I'm not sure if we want this logic in the unit test:
1. for initial local use, assuming the default ports are available and throw an error when not seems OK;
2. if running in Twitter's container environment, the port range is specified by the container and unlikely to be the one(s) specified in the code here.

I think it might be easier to leave the server setup to a separate script. To summarize (including the comment on socket IO):
- a module that deals with all client-side network operations;
- a module that sets up the server, which takes executable location and port as input;
- an "orchestrator" that provides info to the server module and then passes the location of the server to the client module.